launcher-implementation/src/main/scala/xsbt/boot/Pre.scala

```diff
@@ -81,6 +81,8 @@ object Pre {
   final val isWindows: Boolean = System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")
   final val isCygwin: Boolean = isWindows && java.lang.Boolean.getBoolean("sbt.cygwin")
 
+  private[boot] def fileFromPath(path: String): File =  new File(path.replaceFirst("^~", System.getProperty("user.home")))
```

Comment:
`~` alone may refer to other user's home directory, so if we are doing substitution it should only be done for `~/file.txt` but not `~someone/file.txt`.
Also could we rename the function as `substituteTilde` plz?